src/dataregistry/query.py

```diff
@@ -85,32 +85,28 @@ class Query:
     Class implementing supported queries
     """
 
-    def __init__(self, db_engine, dialect, schema_version=SCHEMA_VERSION):
+    def __init__(self, db_connection):
         """
         Create a new Query object. Note this call should be preceded
-        by a call to create_db_engine, which will return values for
-        db_engine and dialect
+        by creation of a DbConnection object
 
         Parameters
         ----------
-        db_engine :   sqlalchemy engine object
-        dialect : str
-            identifies target db type (e.g. 'postgresql')
-        schema_version : str
-            Which database schema to connect to.
-            Current default is 'registry_beta'
+        db_connection : DbConnection object
+            Encompasses sqlalchemy engine, dialect (database backend)
+            and schema version
         """
-        self._engine = db_engine
-        self._dialect = dialect
-        if dialect == "sqlite":
+        self._engine = db_connection.engine
+        self._dialect = db_connection.dialect
+        if self._dialect == "sqlite":
```

Comment:
Is this needed? Is this not repeating what is done when the DbConnection class is created?

Just need `self._schema_version = db_connection.schema`?

Reply:
Right.